### PR TITLE
apply black -l 75 to all code

### PR DESCRIPTION
**Pull request by prjemian**
All code commits must pass flake8 (already a CI requirement).  Apply `black -l 75` code styling to all python files (except vendor code such as versioneer files).

### CODE REVIEW

**Comment by prjemian** [comment]
Also apply some protections against the aggressive styling of `black`, using `# fmt: off` & `#fmt: on` boundaries.  Sometimes, the source code is written so that a human might read it.

**Comment by prjemian** [comment]
@mrakitin : ready for review

File: .flake8 [hunk 1]
```diff
@@ -5,6 +5,7 @@ exclude =
     build,
     dist,
     versioneer.py,
-    docs/source
+    docs/source,
+    hkl/_version.py
 max-line-length = 115
```
Comment (mrakitin) [inline .flake8:10]: The flake8 boundary is 115 symbols. Would it make sense to coordinate it with `black`'s configuration?

**Review by mrakitin — COMMENTED** [review]
I wonder if we should rather keep a more-restricted line length of 115 symbols? That's done, e.g. in https://github.com/bluesky/bluesky-queueserver/blob/main/pyproject.toml, and we run the [linter action](https://github.com/bluesky/bluesky-queueserver/blob/main/.github/workflows/black.yml) there to enforce it (it will use the same `pyproject.toml` configuration).

I am not against the 75-symbol limitation, just would make sense to keep these number in sync between `flake8` and `black`. So, in addition to the 75- and 115-symbol max line length, we have two more options:
- 79 ([PEP8 recommendation](https://www.python.org/dev/peps/pep-0008/#maximum-line-length))
- 120 ([default in Pycharm](https://stackoverflow.com/a/17319775))

What should we pick?

File: .flake8 (hunk 1 above)
Comment (prjemian) [reply .flake8:10]: 75 is my arbitrary choice based on screen readbility.  If we use 115 in other bluesky packages, then let's go with the flow.

Comment (prjemian) [reply .flake8:10]: read*a*bility

**Comment by prjemian** [comment]
@mrakitin The Github Actions workflow for `black` applies the default line length, not `-l 115`.  I suggest we go with that.

```
(bluesky_2021_1) prjemian@poof ~/.../Bluesky/hklpy $ black --help
Usage: black [OPTIONS] [SRC]...

  The uncompromising code formatter.

Options:
  -c, --code TEXT                 Format the code passed in as a string.
  -l, --line-length INTEGER       How many characters per line to allow.
                                  [default: 88]
...
```

**Comment by prjemian** [comment]
We can keep the 115 line length: see https://black.readthedocs.io/en/stable/pyproject_toml.html

**Comment by prjemian** [comment]
The 1black .` test passes locally but continues to fail on GH Actions.  Likely to remove that action until we understand it better and can control it.

**Comment by prjemian** [comment]
Example: https://github.com/bluesky/hklpy/pull/105/checks?check_run_id=1736316333#step:5:358

Will disable the `black.yml` workflow for now.  Can still apply the test locally but enforcement via GH Actions is problematic.

**Comment by prjemian** [comment]
@mrakitin Ready for review

**Comment by mrakitin** [comment]
> @mrakitin The Github Actions workflow for `black` applies the default line length, not `-l 115`. I suggest we go with that.
> 
> ```
> (bluesky_2021_1) prjemian@poof ~/.../Bluesky/hklpy $ black --help
> Usage: black [OPTIONS] [SRC]...
> 
>   The uncompromising code formatter.
> 
> Options:
>   -c, --code TEXT                 Format the code passed in as a string.
>   -l, --line-length INTEGER       How many characters per line to allow.
>                                   [default: 88]
> ...
> ```

As mentioned above, it uses such a file for the configuration: https://github.com/bluesky/bluesky-queueserver/blob/main/pyproject.toml. With that configuration, the linter action should work fine.

And, now we have another, `black`'s default of 88 symbols.

**Comment by mrakitin** [comment]
> We can keep the 115 line length: see https://black.readthedocs.io/en/stable/pyproject_toml.html

Oops, did not read this comment. I see you added it in the following commits.

**Comment by mrakitin** [comment]
When I run it locally, I still see those 7 files changed:
```bash
$ black .
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/docs/source/conf.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/diffract.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/engine.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/tests/test_diffract.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/util.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/calc.py
reformatted /Users/mrakitin/src/mrakitin/DAMA/hklpy/hkl/tests/test_tardis.py
All done! ✨ 🍰 ✨
7 files reformatted, 10 files left unchanged.
$ black --version
black, version 20.8b1
```

What's different on your side, @prjemian?

**Comment by prjemian** [comment]
```
(bluesky_2021_1) prjemian@poof ~/Documents/raw $ black --version
black, version 19.10b0
(bluesky_2021_1) prjemian@poof ~/Documents/raw $ python --version
Python 3.8.2
```

I'll try upgrading black

**Comment by prjemian** [comment]
My version of `black` is conda-installed.  No updates proposed for `black` with `conda update black -c defaults -c conda-forge -c nsls2forge` so updated via pip:
```
(bluesky_2021_1) prjemian@poof ~/Documents/raw $ pip install -U black
Requirement already satisfied: black in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (19.10b0)
Collecting black
  Downloading black-20.8b1.tar.gz (1.1 MB)
     |████████████████████████████████| 1.1 MB 793 kB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: typed-ast>=1.4.0 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (1.4.2)
Requirement already satisfied: toml>=0.10.1 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (0.10.1)
Requirement already satisfied: pathspec<1,>=0.6 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (0.7.0)
Requirement already satisfied: appdirs in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (1.4.4)
Requirement already satisfied: typing-extensions>=3.7.4 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (3.7.4.3)
Requirement already satisfied: regex>=2020.1.8 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (2020.11.13)
Requirement already satisfied: click>=7.1.2 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (7.1.2)
Requirement already satisfied: mypy-extensions>=0.4.3 in /home/prjemian/Apps/anaconda/envs/bluesky_2021_1/lib/python3.8/site-packages (from black) (0.4.3)
Building wheels for collected packages: black
  Building wheel for black (PEP 517) ... done
  Created wheel for black: filename=black-20.8b1-py3-none-any.whl size=124184 sha256=8e790cca456afa2ea0def41b82e66f5ff9e069b4e58f3a96a5c99686f66a9609
  Stored in directory: /home/prjemian/.cache/pip/wheels/95/a4/59/10cd5378d52f92cdb45025f040e4686e10ae5217961c25fd66
```

```
(bluesky_2021_1) prjemian@poof ~/Documents/raw $ black --version
black, version 20.8b1
(bluesky_2021_1) prjemian@poof ~/Documents/raw $ cd ../projects/Bluesky/hklpy/
(bluesky_2021_1) prjemian@poof ~/.../Bluesky/hklpy $ make black
black .
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/docs/source/conf.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/diffract.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/engine.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/tests/test_diffract.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/util.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/calc.py
reformatted /home/prjemian/Documents/projects/Bluesky/hklpy/hkl/tests/test_tardis.py
All done! ✨ 🍰 ✨
7 files reformatted, 10 files left unchanged.
```

Matches your numbers above now.  Yikes such dependencies!  Seems `black` is not sufficiently stable for us to use it in CI workflows.

**Comment by mrakitin** [comment]
Yeah, that's surprising the close versions produce such different results.

As the fixes have been implemented, I believe we can re-enable the GitHub Actions config for `black` now. Let's see if it fails or succeeds.

**Comment by prjemian** [comment]
@mrakitin You were right!  Success now.

**Review by mrakitin — CHANGES_REQUESTED** [review]
`black.yml` was completely removed in the last commit. I believe you intended to move it instead.

**Pull request merged**